### PR TITLE
Converts ALL typepaths to lowercase

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

From now on all paths are required to use snake case.
Replaces all capital letters in typepaths with lowercase ones. Converts CamelCase to snake_case. Renames some paths that are hard to read.
Changes
``/obj/item/reagent_container/pill/C2`` to ``/obj/item/reagent_container/pill``
and 
``/obj/item/storage/pill_bottle/C2`` to ``/obj/item/storage/pill_bottle``
since those paths don't do anything.

Regex used to find these: ``
(?<![^\n])/((mob|datum|atom|obj|area)([^\s^\(^/]|/(?!(/|var/|proc/|\*|#)))*[A-Z])(?!\S*\()``
Not aware of comments. Automatic detection should probably happen with spacemanDMM, no idea how that works though. Will try my best.

## Why It's Good For The Game

It's in the guidelines.
conflicts with https://github.com/tgstation/tgstation/pull/51634

### CODE REVIEW

**Comment by Fikou** [comment]
DONT CHANGE IAN

**Comment by TerraGS** [comment]
My favorite kind of code improvement next to performance boosts.

**Comment by Rohesie** [comment]
Blessed.

File: code/game/objects/items/storage/firstaid.dm [hunk 1]
```diff
@@ -378,13 +378,13 @@
 	for(var/i in 1 to 3)
 		new /obj/item/reagent_containers/pill/potassiodide(src)
 
-/obj/item/storage/pill_bottle/C2/probital
+/obj/item/storage/pill_bottle/probital
```
Comment (ShizCalev) [inline code/game/objects/items/storage/firstaid.dm:381]: Undocumented

**Comment by optimumtact** [comment]
If we're going to standardise we should have lints that enforce it and some documentation

**Comment by MrPerson** [comment]
I notice your regex is missing areas.

File: code/game/objects/items/storage/firstaid.dm (hunk 1 above)
Comment (nemvar) [reply code/game/objects/items/storage/firstaid.dm:381]: Undocumented but no unintentional. Change does not alter any behaviour.

**Comment by nemvar** [comment]
I'm working on the linting but what exactly do you want documented? There is a single actual change in here which is removing the /C2 paths.

**Comment by ATH1909** [comment]
based

**Comment by ATH1909** [comment]
make sure that the DisIntegrate() proc (it's swarmer-related, IIRC) doesn't break because of this PR

**Comment by optimumtact** [comment]
making it clear snake case is the required method naming style

File: code/modules/vending/modularpc.dm [hunk 2]
```diff
@@ -19,7 +19,7 @@
 	premium = list(/obj/item/computer_hardware/card_slot = 2,
 		           /obj/item/computer_hardware/ai_slot = 2,
 		           /obj/item/computer_hardware/printer/mini = 2,
-		           /obj/item/computer_hardware/recharger/APC = 2,
+		           /obj/item/computer_hardware/recharger/apc = 2,
```
Comment (ShizCalev) [inline code/modules/vending/modularpc.dm:22]: if you wouldn't mind, give this a slightly longer name while you're at it. apc_recharger or something like that would be awesome.

**Review by spookydonut — CHANGES_REQUESTED** [review]
why not add your detection regex to `/tools/travis/check_grep.sh`

**Comment by ShizCalev** [comment]
@MrStonedOne @Cyberboss need feedback to ensure sql and tgs3/4 don't get fucked up by this since SQL's unit test keeps failing (ie changing DBQuery to db_query )

**Comment by ShizCalev** [comment]
@MrStonedOne @Cyberboss @SpaceManiac  might need feedback to ensure sql / rust_g.dll and tgs3/4 don't get fucked up by this since mySQL's unit test keeps failing (ie changing DBQuery to db_query )

**Comment by MrStonedOne** [comment]
todo: update server_side_modifications.dm on the game servers. (oranges, this should now be setup to allow editing with admin rights)

**Comment by SpaceManiac** [comment]
rust-g should be fine as its `rust_g.dm` isn't touched. I don't know anything about how TGS or Postgres relate to this

**Comment by Cyberboss** [comment]
TGS uses snake case, so it should be fine. I don't see it in here.

**Review by ShizCalev — APPROVED** [review]

**Review by Rohesie — APPROVED** [review]

**Comment by optimumtact** [comment]
Approving PR's does nothing, merging them is the only valuable feedback

**Comment by spookydonut** [comment]
@Rohesie @optimumtact didn't have the regex added to check_grep.sh, shouldnt have been merged

**Comment by SpaceManiac** [comment]
> didn't have the regex added to check_grep.sh, shouldnt have been merged

not a blocker IMO, certainly a nice-to-have tho

**Comment by optimumtact** [comment]
@spookydonut if you dont' want your PR merged, mark it as draft

**Pull request merged**